pkg/scheduler/plugins/deviceshare/device_cache.go

```diff
@@ -0,0 +1,338 @@
+	}
+	if add {
+		n.deviceUsed[deviceType][int(allocation.Minor)] = quotav1.Add(
+			n.deviceUsed[deviceType][int(allocation.Minor)],
+			allocation.Resources)
+	} else {
+		n.deviceUsed[deviceType][int(allocation.Minor)] = quotav1.SubtractWithNonNegativeResult(
+			n.deviceUsed[deviceType][int(allocation.Minor)],
+			allocation.Resources)
+	}
+}
+
+func (n *nodeDevice) updateAllocateSet(allocations apiext.DeviceAllocations, pod *corev1.Pod, add bool) {
+	podNamespacedName := types.NamespacedName{
+		Namespace: pod.Namespace,
+		Name:      pod.Name,
+	}
+	for deviceType, v := range allocations {
+		if v == nil || len(v) == 0 {
+			break
+		}
+		if n.allocateSet[deviceType] == nil {
+			n.allocateSet[deviceType] = make(map[types.NamespacedName]struct{})
+		}
+		if add {
+			n.allocateSet[deviceType][podNamespacedName] = struct{}{}
+		} else {
+			delete(n.allocateSet[deviceType], podNamespacedName)
+		}
+	}
+}
+
+func (n *nodeDevice) tryAllocateDevice(podRequest corev1.ResourceList) (apiext.DeviceAllocations, []error) {
+	allocateResult := make(apiext.DeviceAllocations)
+	var errors []error
+
+	for deviceType := range deviceResourceNames {
+		switch deviceType {
+		case schedulingv1alpha1.RDMA, schedulingv1alpha1.FPGA:
+			if !hasDeviceResource(podRequest, deviceType) {
+				break
+			}
+			if err := n.tryAllocateCommonDevice(podRequest, deviceType, allocateResult); err != nil {
+				errors = append(errors, err)
+			}
+		case schedulingv1alpha1.GPU:
+			if !hasDeviceResource(podRequest, deviceType) {
+				break
+			}
+			if err := n.tryAllocateGPU(podRequest, allocateResult); err != nil {
+				errors = append(errors, err)
+			}
+		default:
+			klog.Warningf("device type %v is not supported yet", deviceType)
+		}
+	}
+
+	return allocateResult, errors
+}
+
+func (n *nodeDevice) tryAllocateCommonDevice(podRequest corev1.ResourceList, deviceType schedulingv1alpha1.DeviceType, allocateResult apiext.DeviceAllocations) error {
+	podRequest = quotav1.Mask(podRequest, deviceResourceNames[deviceType])
+	nodeDeviceTotal := n.deviceTotal[deviceType]
+	if len(nodeDeviceTotal) <= 0 {
+		return fmt.Errorf("node does not have enough %v", deviceType)
+	}
+
+	var deviceAllocations []*apiext.DeviceAllocation
+
+	if isMultipleCommonDevicePod(podRequest, deviceType) {
+		var commonDeviceWanted int64
+		var podRequestPerCard corev1.ResourceList
+		switch deviceType {
+		case schedulingv1alpha1.RDMA:
+			commonDevice := podRequest[apiext.KoordRDMA]
+			commonDeviceWanted = commonDevice.Value() / 100
+			podRequestPerCard = corev1.ResourceList{
+				apiext.KoordRDMA: *resource.NewQuantity(commonDevice.Value()/commonDeviceWanted, resource.DecimalSI),
+			}
+		case schedulingv1alpha1.FPGA:
+			commonDevice := podRequest[apiext.KoordFPGA]
+			commonDeviceWanted = commonDevice.Value() / 100
+			podRequestPerCard = corev1.ResourceList{
+				apiext.KoordFPGA: *resource.NewQuantity(commonDevice.Value()/commonDeviceWanted, resource.DecimalSI),
```

Comment:
commonDevice.Value()/commonDeviceWanted,  please leave a space before&after the operator ,  like commonDevice.Value() / commonDeviceWanted.

Reply:
I'm with you on that. However, this is automatically formatted by `go fmt`.